segmentstore/storage/impl/src/main/java/io/pravega/segmentstore/storage/impl/bookkeeper/ZooKeeperServiceRunner.java

```diff
@@ -81,15 +82,29 @@ public void start() throws Exception {
     }
 
     public void stop() {
-        ZooKeeperServer zs = this.server.getAndSet(null);
-        if (zs != null) {
-            zs.shutdown();
+        try {
```

Comment:
The changes in this class are unrelated to the issue, please remove it and address it in a separate issue/PR if needed.

Reply:
yeah, I was in the middle of a rebase and couldn't use stash to push this aside.  Will pull it out.